src/framework/application.js

```diff
@@ -750,7 +748,7 @@ class Application extends EventHandler {
         const material = new StandardMaterial();
         material.name = "Default Material";
         material.shadingModel = SPECULAR_BLINN;
-        DefaultMaterial.add(this.graphicsDevice, material);
+        setDefaultMaterial(this.graphicsDevice, material);
```

Comment:
it's done this way to avoid circular dependency few months ago, so I don't want to add it back in again.

Reply:
Oh I see, `Material` itself accesses `getDefaultMaterial()`. I think that's probably the dependency that should be broken.

Instead of material setting the default material on `MeshInstance` itself (in `Material.destroy`), Material can call some function on `MeshInstance` to do the work, like say `onMaterialDestroyed()` or similar.

This would mean `MeshInstance` has dependency on `getDefaultMaterial()` instead of `Material`. Then `getDefaultMaterial()` can implement the cache miss handler and `Application` can be freed from default materials _and_ we can get rid of `setDefaultMaterial()` entirely.

Reply:
I think this is probably the right PR to restructure the material dependencies, but also happy to do if after if you feel strongly about it.

Reply:
I'd prefer separate PR. This PR is focused on creating a device cache for the few use cases, not refactoring material dependencies.